Neos.ContentRepository.Core/Classes/Projection/ContentGraph/VisibilityConstraints.php

```diff
@@ -14,44 +14,51 @@
 
 namespace Neos\ContentRepository\Core\Projection\ContentGraph;
 
+use Neos\ContentRepository\Core\Feature\SubtreeTagging\Dto\SubtreeTag;
+use Neos\ContentRepository\Core\Feature\SubtreeTagging\Dto\SubtreeTags;
+
 /**
- * The context parameters value object
+ * The visibility constraints define a context in which the content graph is accessed.
  *
- * Maybe future: "Node Filter" tree or so as replacement of ReadNodePrivilege?
+ * For example: In the `frontend` context, nodes with the `disabled` tag are excluded. In the `backend` context {@see self::withoutRestrictions()} they are included
  *
  * @api
  */
-final class VisibilityConstraints implements \JsonSerializable
+final readonly class VisibilityConstraints implements \JsonSerializable
 {
-    protected bool $disabledContentShown = false;
-
-    private function __construct(bool $disabledContentShown)
-    {
-        $this->disabledContentShown = $disabledContentShown;
+    /**
+     * @param SubtreeTags $tagConstraints A set of {@see SubtreeTag} instances that will be _excluded_ from the results of any content graph query
+     */
+    private function __construct(
+        public SubtreeTags $tagConstraints,
+    ) {
     }
 
     public function isDisabledContentShown(): bool
     {
-        return $this->disabledContentShown;
+        return $this->tagConstraints->contain(SubtreeTag::disabled());
```

Comment:
@bwaidelich 

I just stumbled upon this one. I don't know if `isDisabledContentShown` is still used anywhere, but the semantics strike me as reversed. Shouldn't this be:
```php
return !$this->tagConstraints->contain(SubtreeTag::disabled());
```

Given that `VisibilityConstraints::frontend()` explicitly contains the tag and `VisibilityConstraints::withoutRestrictions()` doesn't?

Reply:
What a great catch!
I'll check tomorrow, thank you!

Reply:
You were right, obviously.
I didn't realize because this method wasn't used (apart from some outdated code in the PostgreSQL adapter).

I removed it with #4956 Thanks!